astropy/io/ascii/daophot.py

```diff
@@ -165,15 +195,105 @@ class DaophotData(core.BaseData):
     start_line = 0
     comment = r'\s*#'
 
+    def __init__(self):
+        core.BaseData.__init__(self)
+        self.is_multiline = False
+
+    def get_data_lines(self, lines):
+
+        # Special case for multiline daophot databases. Extract the aperture values from the first multiline data block
+        if self.is_multiline:
+            #grab the first column of the special block (aperture values) and recreate the aperture description string
+            aplist = next( zip(*map(str.split, self.first_block)) )
+            self.header.aperture_values = tuple(map( float, aplist ))
+
+        # Set self.data.data_lines to a slice of lines contain the data rows
+        core.BaseData.get_data_lines(self, lines)
+
 
 class DaophotInputter(core.ContinuationLinesInputter):
-    no_continue = r'\s*#'
+
+    continuation_char = '\\'
+    multiline_char = '*'
+    replace_char = ' '
+    re_multiline     = re.compile( r'(#?)[^\\*#]*(\*?)(\\*) ?$' )
+
+
+    def search_multiline(self, lines, depth=150):
+        '''search lines for special continuation character to determine number of continued rows in a
+        datablock.  For efficiency, depth gives the upper limit of lines to search.'''
```

Comment:
Do you know how critical performance is in this case? I rarely read more than a dozen DAOPhot files and I would not care if it takes a few seconds longer per file. On the other hand, my files are not big, with only a few hundred objects or so. On the other hand, I've seen DAOPhot files with close to 100 header lines. How much performance would we loose if we set this to e.g. 1000 or any number so large that we are a factor of a few above the usual number of header lines?